tests/test_web_response.py

```diff
@@ -624,6 +624,23 @@ async def write_headers(status_line, headers):
     assert resp.content_length is None
 
 
+@pytest.mark.parametrize("status", (100, 101, 204, 304))
+async def test_rm_transfer_encoding_rfc_9112_6_3_http_11(status: int) -> None:
+    """Remove transfer encoding for RFC 9112 sec 6.3 with HTTP/1.1."""
+    writer = mock.Mock()
+
+    async def write_headers(status_line, headers):
+        assert hdrs.CONTENT_LENGTH not in headers
+        assert hdrs.TRANSFER_ENCODING not in headers
+
+    writer.write_headers.side_effect = write_headers
```

Comment:
I feel like we should be able to check this on the resp object normally?

But, either way, this check shouldn't be done for 304.

Reply:
> I feel like we should be able to check this on the resp object normally?
Adjusted in 8f9f7527c455863e254fc7e3c7938f84dedee278 Kept the mock since it needs to be an async mock

> But, either way, this check shouldn't be done for 304.

https://github.com/nodejs/llhttp/pull/256#issuecomment-1782909852
> 
> > Transfer-Encoding MAY be sent in a response to a HEAD request or in a 304 (Not Modified) response ([Section 15.4.5](https://www.rfc-editor.org/rfc/rfc9110#section-15.4.5) of [[HTTP](https://datatracker.ietf.org/doc/html/rfc9110)]) to a GET request, neither of which includes a message body, to indicate that the origin server would have applied a transfer coding to the message body if the request had been an unconditional GET. This indication is not required, however, because any recipient on the response chain (including the origin server) can remove transfer codings when they are not needed.

> It is allowed to send transfer-encoding header field in 304 response, but it must be ignored because the message body must be empty.

I think we should strip it out because it will confuse the older clients if we don't

Reply:
Hmm, I guess it's a problem if we have clients that trip up on such a message. But, ideally, it could be used to give additional information about the response... I think the Content-Length header is probably more useful than the Transfer-Encoding header though.

Reply:
> Adjusted in [8f9f752](https://github.com/aio-libs/aiohttp/commit/8f9f7527c455863e254fc7e3c7938f84dedee278) Kept the mock since it needs to be an async mock

Would be better if you can successfully create the mock with `mock.create_autospec(Writer, spec_set=True, instance=True)`

Reply:
b8d5991dea9f1754ae8f5d304692aecdd0e987c6